File: NewsDaily/app/src/main/java/com/andruid/magic/newsdaily/util/DataBindingAdapter.java
package com.andruid.magic.newsdaily.util;

import android.text.Html;
import android.widget.TextView;

import androidx.databinding.BindingAdapter;

import com.squareup.picasso.Picasso;

import java.text.DateFormat;

import de.hdodenhof.circleimageview.CircleImageView;

public class DataBindingAdapter {
    @BindingAdapter({"dateFromMs"})
    public static void formatDate(TextView textView, long ms){
        DateFormat dateFormat = DateFormat.getDateInstance();
        String date = dateFormat.format(ms);
        textView.setText(date);
    }

    @BindingAdapter({"readMore"})
    public static void readMoreUrl(TextView textView, String url){
        String str = "<u>" + url + "</u>";
        textView.setText(Html.fromHtml(str));
    }

    @BindingAdapter({"imageRes"})
    public static void loadFlagRes(CircleImageView imageView, int res){
        Picasso.get()
                .load(res)
                .into(imageView);
    }
}

File: NewsDaily/app/src/main/java/com/andruid/magic/newsdaily/ui/util/DataBindingAdapter.kt
package com.andruid.magic.newsdaily.ui.util

import android.widget.TextView
import androidx.core.text.HtmlCompat
import androidx.databinding.BindingAdapter
import com.squareup.picasso.Picasso
import de.hdodenhof.circleimageview.CircleImageView
import java.text.DateFormat

object DataBindingAdapter {
    @JvmStatic
    @BindingAdapter("dateFromMs")
    fun formatDate(textView: TextView, ms: Long) {
        val dateFormat = DateFormat.getDateInstance()
        val date = dateFormat.format(ms)
        textView.text = date
    }

    @JvmStatic
    @BindingAdapter("readMore")
    fun readMoreUrl(textView: TextView, url: String) {
        val str = "<u>$url</u>"
        textView.text = HtmlCompat.fromHtml(str, HtmlCompat.FROM_HTML_MODE_LEGACY)
    }

    @JvmStatic
    @BindingAdapter("imageRes")
    fun loadFlagRes(imageView: CircleImageView, res: Int) =
        Picasso.get()
            .load(res)
            .into(imageView)
}